roles/lib_openshift/library/docker_creds.py

```diff
@@ -0,0 +1,207 @@
+            module.fail_json(**result)
+        try:
+            j = json.loads(data)
+        # Invalid json data.
+        except Exception as e:
+            result = {'failed': True,
+                    'changed': False,
+                    'msg': str(e),
+                    'state': 'unknown'}
+            module.fail_json(**result)
+        return j
+    else:
+        return {}
+
+def update_config(docker_config, registry, username, password):
+    '''Add our registry auth credentials into docker_config dict'''
+
+    # Add anything that might be missing in our dictionary
+    if 'auths' not in docker_config:
+        docker_config['auths'] = {}
+    if registry not in docker_config['auths']:
+        docker_config['auths'][registry] = {}
+
+    # base64 encode our username:password string
+    encoded_data = base64.b64encode('{}:{}'.format(username, password))
+
+    # check if the same value is already present for idempotency.
+    if 'auth' in docker_config['auths'][registry]:
+        existing_data = docker_config['auths'][registry]['auth']
+        if existing_data == encoded_data:
+            # No need to go further, everything is already set in file.
+            return 0
+    docker_config['auths'][registry]['auth'] = encoded_data
+    return 1
+
+def write_config(module, docker_config, dest):
+    '''Write updated credentials into dest/config.json'''
+    conf_file_path = os.path.join(dest, 'config.json')
+    try:
+        with open(conf_file_path, 'w') as f:
+            json.dump(docker_config, f, indent=8)
+    except Exception as e:
+        result = {'failed': True,
+                'changed': False,
+                'msg': str(e),
+                'state': 'unknown'}
+        module.fail_json(**result)
+
+
+
+def run_module():
+    '''Run this module'''
+    module_args = dict(
+        path=dict(aliases=['dest', 'name'], required=True, type='path'),
+        registry=dict(type='str', required=True),
+        username=dict(type='str', required=True),
+        password=dict(type='str', required=True, no_log=True)
+    )
+
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=False
+    )
+
+    # First, create our dest dir if necessary
+    dest = module.params['path']
+    registry = module.params['registry']
+    username = module.params['username']
+    password = module.params['password']
+
+
+    if not check_dest_dir_exists(module, dest):
+        create_dest_dir(module, dest)
+        docker_config = {}
+    else:
+        # We want to scrape the contents of dest/config.json
+        # in case there are other registries/settings already present.
+        docker_config = load_config_file(module, dest)
+
+    # Put the registry auth info into the config dict.
+    changed = update_config(docker_config, registry, username, password)
+
+    if changed:
+        write_config(module, docker_config, dest)
+        result = {'changed': True}
```

Comment:
This should reflect the changed value and set before.  This reduces the need for the else.
```
result = {'changed': changed}
```